lib/chef/run_list/run_list_expansion.rb

```diff
@@ -131,6 +138,11 @@ def errors
         @missing_roles_with_including_role.map {|item| item.first }
       end
 
+      def parse_into_json
+        @seen_items = {:recipe => {}, :role => {}}
+        {:id => @environment, :run_list => convert_run_list_trace('top level')}.to_json
+      end
```

Comment:
This should probably be `def to_json` and we don't want to be using the `Hash#to_json` monkeypatch from json and want to use `Chef::JSONCompat.to_json()` instead.

There's examples of the pattern to follow in most of the Chef API objects, an example that is roughly similar in complexity using a `#to_hash` method inside of `#to_json`:

https://github.com/chef/chef/blob/master/lib/chef/api_client.rb#L109-L131